### PR TITLE
Doc improvements for v4

### PR DESCRIPTION
**Pull request by vietj**
We are missing a couple of key elements such as putting more emphasis on Futures which are one of the important v4 theme.

### CODE REVIEW

File: src/main/asciidoc/index.adoc [hunk 1]
```diff
@@ -199,6 +199,25 @@ Vert.x will also provide stack traces to pinpoint exactly where the blocking is
 If you want to turn off these warnings or change the settings, you can do that in the
 {@link io.vertx.core.VertxOptions} object before creating the Vertx object.
 
+== Async results
+
+Vert.x 4 use futures to model asynchronous results.
```
Comment (jponge) [inline src/main/asciidoc/index.adoc:204]: model -> represent

File: src/main/asciidoc/index.adoc [hunk 2]
```diff
@@ -199,6 +199,25 @@ Vert.x will also provide stack traces to pinpoint exactly where the blocking is
...
+
+Any asynchronous method returns a {@link io.vertx.core.Future} object for the result of the call:
+a _success_ or a _failure_.
+
+You cannot interact directly with a future instance, instead you need to set a handler that will be called when the future
```
Comment (jponge) [inline src/main/asciidoc/index.adoc:209]: It's not that you can't interact with a Future. It's the result you can't directly interact with ;-)

File: src/main/asciidoc/index.adoc [hunk 3]
```diff
@@ -199,6 +199,25 @@ Vert.x will also provide stack traces to pinpoint exactly where the blocking is
...
+completes, like any other kind of event.
+
+[source,$lang]
+----
+{@link examples.CoreExamples#exampleFuture1}
+----
+
+NOTE: Vert.x 3 provides a callback only model. To allow an easy migration to Vert.x 4 we decided that each asynchronous method
```
Comment (jponge) [inline src/main/asciidoc/index.adoc:217]: callback-only

**Review by jponge — CHANGES_REQUESTED** [review]

File: src/main/asciidoc/index.adoc [hunk 4]
```diff
@@ -218,14 +218,34 @@ NOTE: Vert.x 3 provides a callback only model. To allow an easy migration to Ver
 has also a callback version. The `props` method above has also a version {@link io.vertx.core.file.FileSystem#props(java.lang.String, io.vertx.core.Handler)}
 with a callback as method argument.
 
-== Async coordination
+== Future composition
```
Comment (jponge) [inline src/main/asciidoc/index.adoc:221]: 👍

File: src/main/asciidoc/index.adoc (hunk 3 above)
Comment (vietj) [reply src/main/asciidoc/index.adoc:217]: why ? (I'm curious)

Comment (jponge) [reply src/main/asciidoc/index.adoc:217]: Well there's a `-` for `whatever-only` :-)

**Comment by vietj** [comment]
@jponge added feedback, PTAL :-)

**Pull request merged**